src/compiler/checker.ts

```diff
@@ -9588,6 +9602,14 @@ namespace ts {
 
             const signature = getResolvedSignature(node);
             if (node.expression.kind === SyntaxKind.SuperKeyword) {
+                const containgFunction = getContainingFunction(node.expression);
+
+                if (containgFunction && containgFunction.kind === SyntaxKind.Constructor) {
+                    const nodeLinks = getNodeLinks(containgFunction);
+                    if (!(nodeLinks.flags & NodeCheckFlags.HasSeenThisCall)) {
+                        nodeLinks.flags |= NodeCheckFlags.HasSeenSuperBeforeThis;

```

Comment:
why do we need this flag? can we report the error here? all what we need to know is if the superCall expected or not, and that should be a simple check.. correct?